src/utils/helper-functions/utils.ts

```diff
@@ -11,63 +10,100 @@ import {
 } from './guards'
 
 export const MODULE_PATHS = {
+  componentStore: '@ngrx/component-store',
   effects: '@ngrx/effects',
+  store: '@ngrx/store',
 }
 
-export function findClassDeclarationNode(
-  node: TSESTree.Node,
-): TSESTree.ClassDeclaration | null {
-  if (isClassDeclaration(node)) {
-    return node
+export function getNearestUpperNodeFrom<T extends TSESTree.Node>(
```

Comment:
Instead of having a function for getting a specific type, IMHO we can have a function to lookup given a predicate over parents.